src/flow.jl

```diff
@@ -283,3 +415,423 @@ end
+            update(sw.reservoir, v, sum_at(sw.q0, links_at_node.src[i]) + inflow_wb[i], Δt)
+            sw.q[i] = sw.reservoir.outflow[v]
+        elseif !isnothing(sw.lake) && sw.lake_index[i] != 0
+            v = sw.lake_index[i]
+            update(sw.lake, v, sum_at(sw.q0, links_at_node.src[i]) + inflow_wb[i], doy, Δt)
+            sw.q[i] = sw.lake.outflow[v]
+        else
+            ηsrc = sw.zb[nodes_at_link.src[i]] + sw.h[nodes_at_link.src[i]]
+            ηdst = sw.zb[nodes_at_link.dst[i]] + sw.h[nodes_at_link.dst[i]]
+
+            sw.η_max[i] = max(ηsrc, ηdst)
+            sw.hf[i] = (sw.η_max[i] - sw.zb_max[i])
+
+            if sw.hf[i] > sw.h_thresh
+                sw.a[i] = sw.width_at_link[i] * sw.hf[i] # cross area (rectangular channel)
+                sw.r[i] = sw.a[i] / (sw.width_at_link[i] + 2.0 * sw.hf[i]) # wetted perimeter (rectangular channel)
+                sw.q[i] = local_inertial_riverflow(
+                    sw.q0[i],
+                    ηsrc,
+                    ηdst,
+                    sw.hf[i],
+                    sw.a[i],
+                    sw.r[i],
+                    sw.dl_at_link[i],
+                    sw.mannings_n[i],
+                    sw.g,
+                    sw.froude_limit,
+                    Δt,
+                )
+            else
+                sw.q[i] = 0.0
+            end
+            sw.q_av[i] += sw.q[i] * Δt
+        end
+    end
+    if update_h
+        @threads for i = 1:sw.n
+            if sw.reservoir_index[i] == 0 || sw.lake_index[i] == 0
+                sw.volume[i] =
+                    sw.volume[i] +
+                    (
+                        sum_at(sw.q, links_at_node.src[i]) -
+                        sum_at(sw.q, links_at_node.dst[i]) + sw.inwater[i]
+                    ) * Δt
+                if sw.volume[i] < 0.0
+                    sw.error[i] = sw.error[i] + abs(sw.volume[i])
+                    sw.volume[i] = 0.0 # set volume to zero
+                end
+                sw.volume[i] = max(sw.volume[i] + sw.inflow[i] * Δt, 0.0) # add external inflow
+                sw.h[i] = sw.volume[i] / (sw.dl[i] * sw.width[i])
+                sw.h_av[i] += sw.h[i] * Δt
+            end
+        end
+    end
+end
+
+function update(
+    sw::ShallowWaterRiver,
+    network;
+    inflow_wb = nothing,
+    doy = 0,
+    update_h = true,
+)
+    @unpack nodes_at_link, links_at_node = network
+
+    if !isnothing(sw.reservoir)
+        sw.reservoir.inflow .= 0.0
+        sw.reservoir.totaloutflow .= 0.0
+    end
+    if !isnothing(sw.lake)
+        sw.lake.inflow .= 0.0
+        sw.lake.totaloutflow .= 0.0
+    end
+    sw.q_av .= 0.0
+    sw.h_av .= 0.0
+
+    t = 0.0
+    while t < sw.Δt
+        Δt = stable_timestep(sw)
+        if t + Δt > sw.Δt
+            Δt = sw.Δt - t
+        end
+        shallowwater_river_update(sw, network, Δt, inflow_wb, doy, update_h)
+        t = t + Δt
+    end
+    sw.q_av ./= sw.Δt
+    sw.h_av ./= sw.Δt
+end
```

Comment:
This is not yet done consistently, but a `return sw` would be good here, instead of implicitly returning `sw.h_av`. (Or if you prefer `return nothing`).

Reply:
Will add `return nothing`.